src/main.rs

```diff
@@ -22,72 +42,93 @@ fn main() {
 }
 
 fn _main() -> anyhow::Result<()> {
-    let _ = Cli::parse();
-    let manifest_path =
-        spin_common::paths::resolve_manifest_file_path(spin_common::paths::DEFAULT_MANIFEST_FILE)
-            .context("failed to find spin.toml manifest file")?;
-    let raw_manifest = std::fs::read_to_string(&manifest_path).with_context(|| {
-        format!(
-            "failed to read spin.toml manifest at {}",
-            manifest_path.display()
-        )
-    })?;
-    let manifest = spin_manifest::manifest_from_str(&raw_manifest).with_context(|| {
-        format!(
-            "failed to deserialize spin.toml manifest at {}",
-            manifest_path.display()
-        )
-    })?;
-    if manifest.components.len() > 1 {
-        bail!("Spin applications with more than one component are not yet supported by `spin-test`")
+    match Cli::parse().subcommand.unwrap_or_default() {
+        Subcommand::Run(r) => r.exec(),
     }
-    let component = manifest
-        .components
-        .values()
-        .next()
-        .context("spin.toml did not contain any components")?;
-    let app_path = match &component.source {
-        spin_manifest::schema::v2::ComponentSource::Local(path) => path,
-        spin_manifest::schema::v2::ComponentSource::Remote { .. } => {
-            bail!("components with remote sources are not yet supported by `spin-test`")
+}
+
+#[derive(clap::Parser)]
+struct Run {
+    /// The manifest (spin.toml) file for the application under test.
+    ///
+    /// This may be a manifest file or a directory containing a spin.toml file.
+    #[clap(
+        name = "APP_MANIFEST_FILE",
+        short = 'f',
+        long = "from",
+        alias = "file",
+        default_value = spin_common::paths::DEFAULT_MANIFEST_FILE,
+    )]
+    pub app_source: PathBuf,
```

Comment:
nit: I'm wondering if we should lift this flag up a level to the root command so as to make it common; i'm having a hard time imagining any subcommand for `spin-test` that won't require `app_source`. We can defer this as well until there is apparent commonality.

Reply:
@fibonacci1729 I vaguely recall some clap weirdness with flags declared at root level like they had to come before any subcommand args or something like that.  My memory is vague on the details and I might be misremembering altogether though - it might have only been in specific situations or a completely different crate altogether!  Just flagging it as a thing to double-check if we do lift it up.

Reply:
I'd rather wait on moving flags up to some common level until we're in a situation where we definitely need it.